specification/resources/resource-manager/Microsoft.Resources/preview/2022-08-01-preview/deploymentStacks.json

```diff
@@ -0,0 +1,1353 @@
+              },
+              {
+                "value": "removedBySystem",
+                "description": "Deny assignment has been removed by Azure due to a resource management change (management group move, etc.)"
+              },
+              {
+                "value": "None",
+                "description": "No denyAssignments have been applied."
+              }
+            ]
+          }
+        }
+      }
+    },
+    "ResourceReferenceExtended": {
+      "type": "object",
+      "allOf": [
+        {
+          "$ref": "#/definitions/ResourceReference"
+        },
+        {
+          "$ref": "#/definitions/DeploymentStacksError"
+        }
+      ],
+      "description": "The resource Id extended model."
+    },
+    "DenySettings": {
+      "description": "Defines how resources deployed by the deployment stack are locked.",
+      "type": "object",
+      "properties": {
+        "mode": {
+          "type": "string",
+          "description": "denySettings Mode.",
+          "enum": [
+            "denyDelete",
+            "denyWriteAndDelete",
+            "none"
+          ],
+          "x-ms-enum": {
+            "name": "DenySettingsMode",
+            "modelAsString": true,
+            "values": [
+              {
+                "value": "denyDelete",
+                "description": "Authorized users are able to read and modify the resources, but cannot delete."
+              },
+              {
+                "value": "denyWriteAndDelete",
+                "description": "Authorized users can only read from a resource, but cannot modify or delete it."
+              },
+              {
+                "value": "none",
+                "description": "No denyAssignments have been applied."
+              }
+            ]
+          }
+        },
+        "excludedPrincipals": {
+          "type": "array",
+          "description": "List of AAD principal IDs excluded from the lock. Up to 5 principals are permitted.",
+          "items": {
+            "type": "string"
+          }
+        },
+        "excludedActions": {
+          "type": "array",
+          "description": "List of role-based management operations that are excluded from the denySettings. Up to 200 actions are permitted. If the denySetting mode is set to 'denyWriteAndDelete', then the following actions are automatically appended to 'excludedActions': '*/read' and 'Microsoft.Authorization/locks/delete'. If the denySetting mode is set to 'denyDelete', then the following actions are automatically appended to 'excludedActions': 'Microsoft.Authorization/locks/delete'. Duplicate actions will be removed.",
+          "items": {
+            "type": "string"
+          }
+        },
+        "applyToChildScopes": {
+          "type": "boolean",
+          "description": "DenySettings will be applied to child scopes."
+        }
+      }
+    },
+    "AzureResourceBase": {
```

Comment:
>AzureResourceBase [](http://example.com/codeflow?start=5&length=17)

can you just please use the definition from commontypes instead ? [https://github.com/Azure/azure-rest-api-specs/blob/d632c45f406c48b5f72b7d031bf0742e46246c4b/specification/common-types/resource-management/v5/types.json#L92](https://github.com/Azure/azure-rest-api-specs/blob/d632c45f406c48b5f72b7d031bf0742e46246c4b/specification/common-types/resource-management/v5/types.json#L92)

Reply:
Do we still use this definition if Deployment Stacks does have a location but is still a proxy resource? It says in the Common Type description that "The resource model definition for a Azure Resource Manager proxy resource. It will not have tags and a location" which would be inaccurate for stacks.